plugins/plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/plugins/ProjectExplorerTreeColorizer.java

```diff
@@ -82,7 +83,12 @@ public void onGitStatusChanged(String endpointId, StatusChangedEventDto statusCh
     tree.getNodeStorage()
         .getAll()
         .stream()
-        .filter(node -> node instanceof FileNode)
+        .filter(
+            node ->
+                node instanceof FileNode
+                    && statusChangedEventDto
+                        .getProjectName()
+                        .equals(getRootProject(((ResourceNode) node).getData().asFile()).getName()))
```

Comment:
Cast to `FileNode` since it is already checked that `node instanceof FileNode`

Reply:
done